java/src/jmri/jmrit/logixng/actions/swing/ActionBlockSwing.java

```diff
@@ -0,0 +1,362 @@
+package jmri.jmrit.logixng.actions.swing;
+
+import java.util.List;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.swing.*;
+
+import jmri.Block;
+import jmri.BlockManager;
+import jmri.InstanceManager;
+import jmri.JmriException;
+import jmri.Memory;
+import jmri.MemoryManager;
+import jmri.NamedBeanHandle;
+import jmri.NamedBeanHandleManager;
+import jmri.jmrit.logixng.*;
+import jmri.jmrit.logixng.actions.ActionBlock;
+import jmri.jmrit.logixng.actions.ActionBlock.BlockOperation;
+import jmri.jmrit.logixng.actions.ActionBlock.DirectOperation;
+import jmri.jmrit.logixng.swing.SwingConfiguratorInterface;
+import jmri.jmrit.logixng.util.parser.ParserException;
+import jmri.util.swing.BeanSelectCreatePanel;
+import jmri.util.swing.JComboBoxUtil;
+
+/**
+ * Configures an Block object with a Swing JPanel.
+ *
+ * @author Dave Sand Copyright 2021
```

Comment:
I have forgot to add myself here in the Swing classes. It would be good if you could add me here.
```
* @author Daniel Bergqvist Copyright 2021
* @author Dave Sand Copyright 2021
```